src/ol_infrastructure/components/aws/cache.py

```diff
@@ -0,0 +1,211 @@
+            return auth_token
+        if values['encrypt_transit'] and auth_token is None:
+            raise ValueError('Cannot encrypt transit with no auth token configured')
+        token_valid = 16 <= len(auth_token or '') <= 128 and bool(re.search('[^\'"/@\W]+', auth_token or ''))  # noqa: WPS221, W605, E501
+        if not token_valid:
+            raise ValueError('The configured auth token has invalid characters. '
+                             'Only printable ASCII characters excluding ", / and @ are allowed.')
+        return auth_token
+
+    @validator('cluster_name')
+    def is_valid_cluster_name(cls: 'OLAmazonRedisConfig', cluster_name: Text) -> Text:  # noqa: N805
+        is_valid: bool = True
+        is_valid = 1 < len(cluster_name) < 41
+        is_valid = not bool(re.search('[^a-zA-Z-9-]', cluster_name))
+        if not is_valid:
+            raise ValueError('The cluster name does not comply with the Elasticache naming constraints for Redis')
+        return cluster_name
+
+
+class OLAmazonMemcachedConfig(OLAmazonCacheConfig):
+    engine: Text = 'memcached'
+    engine_version: Text = '1.5.16'
+    port: PositiveInt = PositiveInt(11211)
+    num_instances: conint(ge=1, le=20) = 3  # type: ignore
+
+    @validator('cluster_name')
+    def is_valid_cluster_name(cls: 'OLAmazonMemcachedConfig', cluster_name: Text) -> Text:  # noqa: N805
+        is_valid: bool = True
+        is_valid = 1 < len(cluster_name) < 51
+        is_valid = not bool(re.search('[^a-zA-Z-9-]', cluster_name))
+        if not is_valid:
+            raise ValueError('The cluster name does not comply with the Elasticache naming constraints for Memcached')
+        return cluster_name
+
+
+class OLAmazonCache(pulumi.ComponentResource):
+
+    def __init__(
+            self,
+            cache_config: Union[OLAmazonRedisConfig, OLAmazonMemcachedConfig],
+            opts: pulumi.ResourceOptions = None
+    ):
+        super().__init__(
+            'ol:infrastructure:aws:elasticache:OLAmazonCache',
+            cache_config.cluster_name,
+            None,
+            opts
+        )
+        resource_options = pulumi.ResourceOptions(parent=self).merge(opts)  # type: ignore
+
+        clustered_redis = cache_config.engine == 'redis' and getattr(cache_config, 'cluster_mode_enabled')
+
+        cache_parameters = [elasticache.ParameterGroupParameterArgs(name=name, value=setting)
+                            for name, setting in (cache_config.parameter_overrides or {}).items()]
+        if clustered_redis:
+            cache_parameters.append(
+                elasticache.ParameterGroupParameterArgs(
+                    name='cluster-enabled', value='yes'
+                )
+            )
+            if cache_config.engine_version.startswith('6'):
+                cache_options = resource_options.merge(
+                    pulumi.ResourceOptions(ignore_changes=['engine_version'])
+                )  # type: ignore
```

Comment:
Not sure I follow what this is supposed to do and also am hesitant about having the Redis version hardcoded here.

Reply:
It's not hardcoded, it's just set to a sane default. As to what's happening, the issue is that with releases of Redis >= 6 AWS refers to that version as `6.x`, but when it gets deployed that resolves to a concrete version (e.g. 6.0.5), so that when you attempt to re-run the `up` it shows an attribute mismatch, so this tells Pulumi to ignore that difference.